File: Utils/src/hkccpacmanrobot/utils/map/MapContent.java
package hkccpacmanrobot.utils.map;

/**
 * Created by 13058456a on 3/18/2015.
 */
public class MapContent implements Cloneable{
    public long time, preservedLong;

    public MapContent() {
    }

    public MapContent(long time, long preservedLong) {
        this.time = time;
        this.preservedLong = preservedLong;
    }

    public void set(MapContent mapContent) {
        this.time = mapContent.time;
        this.preservedLong = mapContent.preservedLong;
    }

    @Override
    protected Object clone()  {
        return new MapContent(time,preservedLong);
    }
}

File: src/main/scala-2.11/studentrobot/code/MapContent.scala
package studentrobot.code

/**
 * Created by beenotung on 3/21/15.
 */
class MapContent(var time: Long = 0L, var preservedLong: Long = 0L) {

  def set(mapContent: MapContent) {
    time = mapContent.time
    preservedLong = mapContent.preservedLong
  }

  protected override def clone: AnyRef = {
    new MapContent(time, preservedLong)
  }
}
